### ISSUE TITLE
P3: Include CoderBot Messages in New Message Count Display

### ISSUE THREAD

**Issue by ajefts** [appirio-tech/connect-app#3405]
**Objective:** CodeBot sends a message to a client on a project upon project creation, prompting them with next steps. To facilitate the experience of responsiveness from Topcoder during this critical period, the CoderBot message should be included in the count of new messages a user sees for their project. 

<p></p>

**Requirements:**

- Include CoderBot messages in the count of new messages in a project

- Display count of new messages, including CoderBot messages, on the Messages tab of a project for immediate client visibility to this communication upon project creation.

**Comment by ajefts** [comment]
Link to feature: https://tcprod.productboard.com/feature-board/planning/features/2757140

**Comment by vikasrohit** [comment]
@maxceem can we do it quickly?

**Comment by maxceem** [comment]
I think we can, previously we filtered creating notification from Coder bot, so I guess we should remove this filtering? https://github.com/topcoder-platform/tc-notifications/blob/dev/connect/connectNotificationServer.js#L410-L416

Though, as I understand, removing this filter would also create notifications about topics created in each phase, as we bot also creates a message there.

**Comment by vikasrohit** [comment]
If we remove that filter, I guess it would work fine because we are already differentiating between phase topics and regular topics while showing the notifications counts. Right?

**Comment by maxceem** [comment]
Regarding the counts - yes. But I guess we would still get notifications about the Bot welcome messages in each phase.

**Comment by vikasrohit** [comment]
Yes, that has to happen to get the counts. May be we can add this filter in email service where we don't send notifications for coder bot actions, if emails creates noise. We have to discuss that with product managers.

**Comment by lakshmiathreya** [comment]
cc @nithyaasworld lakshmiathreya

**Comment by nithyaasworld** [comment]
@vikasrohit I am not seeing any new message or message count. Below are the steps followed:

1. created a new project in Test Environment
2. Received an email from Topcoder regarding the project creation
3. But no coder bot message or message count is displayed in the project.

User used: devcustomer02 / appirio123
Project created: https://connect-test01.topcoder-dev.com/projects/8595

Please let me know if I am missing anything. Thanks.

**Comment by vikasrohit** [comment]
This is not moved to production in this release. We would take it in next release.

**Comment by vikasrohit** [comment]
@maxceem can we wrap this one in next release? I am moving this to next release.

**Comment by maxceem** [comment]
> @maxceem can we wrap this one in next release? I am moving this to next release.

Yes, we can handle it for the next release. Though I'd like to clarify the requirements.

If we just enable notifications about posts the coder bot, then we would also get notifications about such posts when we add phases.

1. Should we enable notification about CoderPost only about messages for the project but not for the `phases`?

2. You've noticed that we should add a special filter to not send emails about posts from CoderBot https://github.com/appirio-tech/connect-app/issues/3405#issuecomment-564970600. Should we do this or first do without such filter and see how it works?

**Comment by vikasrohit** [comment]
I think we should just enable all coder bot topic/post creation actions to pass through now and see how it works. I mean, I don't see issue now with generating notifications/emails for phase topics because for most of the project templates we don't create default phases now and phases are added after project is reviewed. However, one use case which we might need to check is that, because customer can not see any phase which is not yet in status planned or above, we don't accidentally send coder bot's action notifications/emails for topics/posts of such phases.

**Comment by vikasrohit** [comment]
@maxceem I just checked with product managers, it seems this feature is on hold, so no need to work on it in next release. Moving to backlog.
